### PR TITLE
onboard etcd for reading config

### LINKED ISSUE #40: Add command for publishing call confirmation to etcd
As a first step to hot reload configuration, we should create new cobra command that will read config and push call config to etcd.

### PR DESCRIPTION
**Pull request by shan-96**
Partially closes #40 

- onboard etcd
- add new command to read call configs and push it to etcd
- will create separate MR to add watcher as this one is already pretty big

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -8,31 +8,64 @@ require (
...
+	google.golang.org/genproto/googleapis/rpc v0.0.0-20240314234333-6e1732d8331c // indirect
+	google.golang.org/grpc v1.62.1 // indirect
+	google.golang.org/protobuf v1.33.0 // indirect
 	gopkg.in/ini.v1 v1.67.0 // indirect
+	gopkg.in/yaml.v2 v2.4.0 // indirect
 	gopkg.in/yaml.v3 v3.0.1 // indirect
+	sigs.k8s.io/yaml v1.2.0 // indirect
 )
+
+replace github.com/coreos/bbolt => go.etcd.io/bbolt v1.3.11 // required for etcd compatibility
```
Comment (shan-96) [inline go.mod:69]: the grpc based newer version of etcd is recommended but it has many dependency issues

https://github.com/etcd-io/etcd/issues/11749#issuecomment-1065146905

File: pkg/cmd/init.go [hunk 2]
```diff
@@ -34,6 +36,8 @@ func InitCommands(build, version string) *cobra.Command {
 	cmd.PersistentFlags().StringVar(&args.configPath, "config", "./runtime/config.yaml", "config file path")
 	cmd.PersistentFlags().StringVar(&args.logLevel, "log-level", "info", "log level (debug, info, warn, error)")
 	cmd.PersistentFlags().BoolVar(&args.textFormat, "log-text", false, "log in text format, otherwise JSON")
+	cmd.PersistentFlags().StringVar(&args.etcdURL, "etcdURL", "localhost:2379", "the host:port for etcd")
```
Comment (shan-96) [inline pkg/cmd/init.go:39]: maybe better to have separate `etcd.conf`

File: pkg/cmd/cfg.go [hunk 3]
```diff
@@ -40,3 +44,33 @@ func initConfig(configPath string) (*config, error) {
...
+		return err
+	}
+
+	etcd := NewEtcdHandler(etcdURL, time.Duration(dialTimeoutSeconds*int(time.Second)))
+
+	err = etcd.Put(ctx, "CallConfig", string(callConfigJSON))
+
+	if err != nil {
+		return err
+	}
```
Comment (ksysoev) [inline pkg/cmd/cfg.go:73]: @shan-96 lets move this logic into call repo, in this case in this case we'll be able to run validation logic before config upload.

File: pkg/cmd/etcd.go [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+	"go.etcd.io/etcd/clientv3"
+)
+
+type Etcd interface {
+	Put(ctx context.Context, key string, value string) error
+
+	// TODO: implement watcher
+	Watch(ctx context.Context, key string)
+}
```
Comment (ksysoev) [inline pkg/cmd/etcd.go:17]: also interfaces should be defined on consumer side, lets move definition of the ConfigStore to Call repo

File: pkg/cmd/init.go [hunk 5]
```diff
@@ -63,3 +67,26 @@ func ServerCommand(arg *args) *cobra.Command {
...
+// The config is loaded and then pushed to etcd for watching changes.
+// It can take cfgPath of type *string which is the path to the configuration file as an argument.
+// It also takes the etcd host URL and dial timeout in seconds as argument
+// It returns a pointer to a cobra.Command which can be executed to load the config.
+// It returns an error if the logger initialization fails, the configuration cannot be loaded, or there is error thrown by etcd.
+func ReadConfigCommand(arg *args) *cobra.Command {
+	return &cobra.Command{
+		Use:   "read-config",
+		Short: "Read config and push call config to etcd",
+		Long:  "Read config and push call config to etcd for hot reloads. Also sets up a watcher for the config",
```
Comment (ksysoev) [inline pkg/cmd/init.go:81]: What do you think if we design commands like:
- `config upload`
- in future we can add `config backup` or `config download`

File: pkg/cmd/init.go (hunk 2 above)
Comment (ksysoev) [reply pkg/cmd/init.go:39]: @shan-96 lets use existing config file.

**Review by ksysoev — APPROVED** [review]

**Pull request merged**